js/directives/lucca/user-picker.js

```diff
@@ -0,0 +1,322 @@
+			var homonymsArray = [];
+
+			loadPropertiesFromUrls(homonyms).then(function (response) {
+								// Add homonyms properties for each user
+								_.each(homonyms, function (user, item) {
+									// Get the returned user
+									var userWithProps = response[item].data.data.items[0];
+
+									// Add each property to the user
+									_.each($scope.properties, function (prop) {
+										var newProp = prop.split('.')[0];
+										user[newProp] = userWithProps[newProp];
+									});
+								});
+							});
+		};
+
+		var loadPropertiesFromUrls = function (homonyms) {
+			var urlCalls = [];
+			var queries = [];
+			var templateQuery = "/api/v3/users?id=";
+			var fields = "&fields=id,name,firstname,lastname,dtcontractend";
+			var deferred = $q.defer();
+
+			// WARNING: Do not check if the properties exist!
+			// WARNING: If they do not exist, the request will fail
+			_.each($scope.properties, function (prop) {
+				fields += "," + prop;
+			});
+
+			_.each(homonyms, function (user) {
+				queries.push(templateQuery + user.id + fields);
+			});
+
+			_.each(queries, function (query) {
+				urlCalls.push($http.get(query));
+			});
+
+			$q.all(urlCalls)
+			.then(
+				function (response) {
+					deferred.resolve(response);
+				},
+				function (error) {
+					deferred.reject(error);
+				}
+				);
+			return deferred.promise;
+		};
+
+		var countTimeout = function (input, delay) {
+			delay = delay || 5000; // default delay value 1s
+
+			// cancel previous timeout
+			if (timeout.count) {
+				console.log("-- cancel previous timeout");
+				$timeout.cancel(timeout.count);
+			}
+
+			// launch new timeout to save in _delay_ 
+			timeout.count = $timeout(function () {
+				doCount(input);
+			}, delay);
+		};
+
+		var doCount = function (input) {
+			var query = "/api/v3/users?name=like," + input + "&fields=collection.count"; // query for count
+
+			delete timeout.count;
+
+			$http.get(query)
+			.then(function (response) {
+					$scope.count = response.data.data.count; // To update with right query
+					_.last($scope.users).overflow = MAX_COUNT + "/" + $scope.count;
+				});
+		};
+
+		$scope.getProperty = function (user, prop) {
+			var propList = prop.split('.');
+			var value = user[_.first(propList)];
+
+			_.each(propList, function (item) {
+				if (value && (item !== _.first(propList))) {
+					value = value[item];
+				}
+			});
+
+			return value;
+		};
+
+		// Used by UserPickerMultiple
+		// Function executed when onSelect is fired
+		$scope.addSelectedUser = function () {
+			$scope.onSelect();
+			selectedUsersCount++;
+			// Update overflow message
+			if ($scope.count > MAX_COUNT) {
+				_.last($scope.users).overflow = (MAX_COUNT - selectedUsersCount) + "/" + $scope.count;
+			}
+		};
+
+		$scope.find = function (input) {
+			if (input) {
+				getUsers(false, input);
+			}
+			// For initialization
+			else {
+				getUsers(true, "");
+			}
+		};
```

Comment:
`$scope.find` must be the conductor, not delegate this role to the method get user

```js
$scope.find = function(clue){
  // reinit timeouts linked to the way we handle pagination and other stuff that can be called later in the treatment
  reInitStuff();

  // fetch the users that satisfy 'clue'
  getUsersAsync(clue)
  .then(function(users){
    // handle pagination if needed
    if(hasPagination(users)){ handlePaginationAsync(); }

    // handle formerEmployees if needed
    if(hasFormerEmployees(users)) { handleFormerEmployees(); }

    // handle homonyms
    if(hasHomonyms()){ handleHomonymsAsync(); }
  },function(message){
    handleError(message);
  });
}
```

on every async method called don't forget to handle the possible rejection. i did not put it here for concision